flight/Modules/Actuator/actuator.c

```diff
@@ -100,24 +101,102 @@ static float motor_mixer[MAX_MIX_ACTUATORS * MIXERSETTINGS_MIXER1VECTOR_NUMELEM]
 static ActuatorSettingsData actuatorSettings;
 static SystemSettingsAirframeTypeOptions airframe_type;
 
-static float curve1[MIXERSETTINGS_THROTTLECURVE1_NUMELEM];
 static float curve2[MIXERSETTINGS_THROTTLECURVE2_NUMELEM];
 
 static MixerSettingsCurve2SourceOptions curve2_src;
 
 // Private functions
 static void actuator_task(void* parameters);
 
-static float scale_channel(float value, int idx);
+static float scale_channel(float value, int idx, bool active_cmd);
 static void set_failsafe();
 
-static float throt_curve(const float input, const float *curve,
-		uint8_t num_points);
 static float collective_curve(const float input, const float *curve,
 		uint8_t num_points);
 
 volatile enum actuator_interlock actuator_interlock = ACTUATOR_INTERLOCK_OK;
 
+struct dshot_command {
+	union {
+		uint32_t value;
+		struct {
+			uint8_t cmd_id;
+			uint8_t num_to_send;
+			uint16_t channel_mask;
+		};
+	};
+};
+
+static volatile struct dshot_command pending_cmd;
+static struct dshot_command cur_cmd;
+
+static uint16_t desired_3d_mask;
+
+#define DSHOT_COMMAND_DONTSPIN 0
+#define DSHOT_COMMAND_NO3DMODE 9
+#define DSHOT_COMMAND_3DMODE   10
+
+#define DSHOT_COUNT_EXCESSIVE 12
+
+static bool actuator_get_dshot_command(bool armed) {
+	/* Finish off any current command. */
```

Comment:
The logic here is, if we have a command in progress, send it no matter what.  This allows a command to be sent at the arming edge and to continue sending.